tests/integration_tests/test_server.py

```diff
@@ -206,3 +207,40 @@
     assert "name" in msg
     assert "description" in msg
     assert "filename" in msg
+
+
+@pytest.mark.parametrize("command", ("game_host", "game_join"))
+async def test_server_ban_prevents_hosting(lobby_server, database, command):
+    """
+    Players who are banned while they are online, should immediately be
+    prevented from joining or hosting games until their ban expires.
+    """
+    player_id, _, proto = await connect_and_sign_in(
+        ('banme', 'banme'), lobby_server
+    )
+    # User successfully logs in
+    await read_until_command(proto, 'game_info')
+
+    async with database.acquire() as conn:
+        await conn.execute(
+            ban.insert().values(
+                player_id=player_id,
+                author_id=player_id,
+                reason="Test live ban",
+                expires_at=func.date_add(
+                    func.now(), text("interval 1000 year")
+                ),
+                level='GLOBAL'
+            )
+        )
```

Comment:
For such things rather make fixtures, everything that configures you database, should be in fixtures.
Like: give_me_ban fixture will insert ban for you :-)

Reply:
As I understand it the ban here has to be given *after* logging in and not during setup, so that you can log in first?
So if it was a fixture you would need to return a user that's logged in and banned, doing both steps in the fixture?
If that's too much for a fixture to do, then maybe it's enough to move that block of code into a helper function (in terms of readability).

Reply:
Maybe i would prefer test like action login, action ban, test message, instead of just putting query into the db.

Also i would like to see the test, where you have one user with ban, another without, and you ban first, but login second one and ask for ban. You will reveal that bug with `and` operator that is above.

Reply:
I can put it in a function. But it needs to happen after the player has logged in, otherwise the ban will simply be checked on login which is not what we're trying to test here.